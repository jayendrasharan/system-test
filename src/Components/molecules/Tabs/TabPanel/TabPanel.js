import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  className: PropTypes.string,
  children: PropTypes.node,
  index: PropTypes.number,
  isSelected: PropTypes.bool,
  inheritedStyles: PropTypes.string,
  tabId: PropTypes.string,
};

const TabPanel = ({
  children,
  className,
  index,
  isSelected,
  tabId,
  ...others
}) => (
  <div
    id={tabId}
    className={className}
    role="tabpanel"
    aria-expanded={isSelected}
    aria-hidden={!isSelected}
    aria-labelledby={`tab_${index}_${tabId}`}
    hidden={!isSelected}
    {...others}
  >
    {children}
  </div>
);

TabPanel.propTypes = propTypes;

TabPanel.defaultProps = {
  className: '',
  inheritedStyles: '',
  index: 0,
  isSelected: false,
  tabId: '',
  children: '',
};

export default TabPanel;
