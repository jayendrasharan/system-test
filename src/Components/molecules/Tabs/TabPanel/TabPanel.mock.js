const hiddenPanelConfig = {
  id: 'tab_1',
  isSelected: false,
  isBorderedStyle: true,
};

const selectedPanelConfig = {
  id: 'tab_2',
  isSelected: true,
  isBorderedStyle: true,
};

export { hiddenPanelConfig, selectedPanelConfig };
