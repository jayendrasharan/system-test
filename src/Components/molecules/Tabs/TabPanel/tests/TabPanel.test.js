import React from 'react';
import { shallow } from 'enzyme';
// import renderer from 'react-test-renderer';

import TabPanel from '../index';

describe('<TabPanel />', () => {
  const props = {
    tabId: 'default-tab',
  };
  let TabPanelComponent = '';
  beforeEach(() => {
    TabPanelComponent = shallow(<TabPanel {...props}>Test</TabPanel>);
  });

  test('should render correctly', () => {
    expect(TabPanelComponent).toMatchSnapshot();
  });
});
