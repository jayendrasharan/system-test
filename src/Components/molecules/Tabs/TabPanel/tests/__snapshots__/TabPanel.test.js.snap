// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabPanel /> should render correctly 1`] = `
<div
  aria-expanded={false}
  aria-hidden={true}
  aria-labelledby="tab_0_default-tab"
  className=""
  hidden={true}
  id="default-tab"
  inheritedStyles=""
  role="tabpanel"
>
  Test
</div>
`;
